config.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	db   *DbConfig
+	http *HttpConfig
+}
+
+type HttpConfig struct {
+	port int
+}
+
+type DbConfig struct {
+	net          string
+	host         string
+	port         int
+	databaseName string
+	user         string
+	password     string
+}
+
+func (c DbConfig) getAddr() string {
+	if c.net == "tcp" {
+		return c.host + ":" + strconv.Itoa(c.port)
+	}
+	return ""
+}
+
+func LoadConfig(env *string) *Config {
+	//load the config file
+	viper.SetConfigName("app")
+	viper.AddConfigPath("config")
+
+	err := viper.ReadInConfig()
+	if err != nil {
+		fmt.Println("Config file (./config/app.[json|yaml|toml]) not found - abort")
```

Comment:
mostly for simplicity and to reduce the number of concepts a beginner needs to grasp